src/CashuWallet.ts

```diff
@@ -30,10 +33,21 @@ class CashuWallet {
 	 * @param mint Cashu mint instance is used to make api calls
 	 */
 	constructor(keys: MintKeys, mint: CashuMint) {
-		this.keys = keys;
+		this._keys = keys;
 		this.mint = mint;
+		this._keysetId = deriveKeysetId(this._keys);
 	}
 
+	get keys(): MintKeys {
+		return this._keys;
+	}
+	set keys(keys: MintKeys) {
+		this._keys = keys;
+		this._keysetId = deriveKeysetId(this._keys);
+	}
+	get keysetId(): string {
```

Comment:
We could also only store the keys, and derive the keyset from the keys each time we want to read it.

Pros:
We don't need to keep the keyset variable in sync

Cons:
Might be computationally intensive

Reply:
True. I'm ok with both ways .You decide.

Reply:
but they shouldn´t get out of sync as the keys are a private property
 - keys can be set in the constructor - we recalc the keysetId 
 - keys can be set by the setter - we recalc the keysetId 
 
 also hashing is 'slow' / computationally intensive

Reply:
i agree 👍 we just have to be careful within the class to not mess it up 😁